### PR TITLE
WRN-16762: Added support for tailwindcss and update dependencies

### PR DESCRIPTION
**Pull request by hong6316**
### Issue Resolved / Feature Added
Added support for tailwindcss and update and unpined dependencies.

### Resolution
All dependencies updated to latest release.
- create
Coerce Node versions with metadata

- pack
Add support for Tailwind

### Links
WRN-16762

### Comments
Enact-DCO-1.0-Signed-off-by: Taeyoung Hong (taeyoung.hong@lge.com)

### CODE REVIEW

File: commands/pack.js [hunk 1]
```diff
@@ -100,13 +100,17 @@ function details(err, stats, output) {
...
 		messages.warnings.length
 	) {
-		console.log(
-			chalk.yellow(
-				'Treating warnings as errors because process.env.CI = true. ' +
-					'Most CI servers set it automatically.\n'
-			)
-		);
-		return new Error(messages.warnings.join('\n\n'));
+		// Ignore sourcemap warnings in CI builds. See #8227 for more info.
```
Comment (MikyungKim) [inline commands/pack.js:103]: Please revert this change.

File: commands/serve.js [hunk 2]
```diff
@@ -35,12 +35,15 @@ process.on('unhandledRejection', err => {
 // As react-dev-utils assumes the webpack production packaging command is
 // "npm run build" with no way to modify it yet, we provide a basic override
 // to console.log to ensure the correct output is displayed to the user.
-console.log = (log => (data, ...rest) =>
-	typeof data === 'undefined'
-		? log()
-		: typeof data === 'string'
-		? log(data.replace(/npm run build/, 'npm run pack-p'), ...rest)
-		: log.call(this, data, ...rest))(console.log);
+console.log = (
```
Comment (MikyungKim) [inline commands/serve.js:38]: I just ignored this in my PR. Let's skip this change.

File: config/webpack.config.js [hunk 3]
```diff
@@ -400,6 +404,9 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
 						mangle: {
 							safari10: true
 						},
+						// Added for profiling in devtools
```
Comment (MikyungKim) [inline config/webpack.config.js:407]: Which PR related to this?

**Review by MikyungKim — COMMENTED** [review]
You should not change the permission of bin/enact.js to 644. Please revert the change.

File: config/webpack.config.js (hunk 3 above)
Comment (hong6316) [reply config/webpack.config.js:407]: It's CRA`s #7737 PR.
While profiling in React, via using react-dom/profiling, it needs to keep its class name & function name.
These options help to keep names, but these options are for React, CRA.
No needs I think.

File: config/webpack.config.js [hunk 4]
```diff
@@ -145,7 +149,7 @@ module.exports = function (env, ilibAdditionalResourcesPath) {
 							],
 							// Adds PostCSS Normalize to standardize browser quirks based on
 							// the browserslist targets.
-							'postcss-normalize',
+							!useTailwind && require('postcss-normalize')(),
```
Comment (MikyungKim) [inline config/webpack.config.js:152]: Does this need () ?

File: config/webpack.config.js (hunk 3 above)
Comment (MikyungKim) [reply config/webpack.config.js:407]: Please remove them.

File: package.json [hunk 5]
```diff
@@ -105,29 +105,32 @@
...
+    "webpack": "^5.64.4",
+    "webpack-dev-server": "^4.6.0"
+
   },
   "optionalDependencies": {
-    "fsevents": "^2.1.3"
+    "fsevents": "^2.3.2"
   },
   "peerDependencies": {
+    "react": ">= 16",
```
Comment (MikyungKim) [inline package.json:133]: 🤔 Why is this added? and why it's 16?

File: package.json [hunk 6]
```diff
@@ -39,64 +39,64 @@
...
+    "@babel/plugin-proposal-export-default-from": "^7.16.7",
+    "@babel/plugin-transform-react-inline-elements": "^7.16.7",
+    "@babel/plugin-transform-runtime": "^7.16.7",
+    "@babel/preset-env": "^7.16.7",
+    "@babel/preset-react": "^7.16.7",
+    "@babel/preset-typescript": "^7.16.7",
+    "@babel/runtime": "^7.16.7",
+    "@enact/dev-utils": "^4.1.4",
+    "@enact/template-sandstone": "^1.0.0",
+    "@pmmmwh/react-refresh-webpack-plugin": "^0.5.3",
```
Comment (MikyungKim) [inline package.json:53]:
```suggestion
    "@pmmmwh/react-refresh-webpack-plugin": "^0.5.4",
```

File: package.json [hunk 7]
```diff
@@ -39,64 +39,64 @@
...
-    "babel-plugin-transform-rename-import": "2.3.0",
-    "case-sensitive-paths-webpack-plugin": "2.4.0",
-    "chalk": "4.1.0",
-    "core-js": "3.6.5",
-    "cross-spawn": "7.0.3",
-    "css-loader": "6.2.0",
-    "css-minimizer-webpack-plugin": "3.0.2",
-    "dotenv": "9.0.2",
-    "dotenv-expand": "5.1.0",
+    "babel-jest": "^27.4.2",
```
Comment (MikyungKim) [inline package.json:60]:
```suggestion
    "babel-jest": "^27.5.1",
```

File: package.json (hunk 5 above)
Comment (hong6316) [reply package.json:133]: It's the same with CRA. from CRA`s #9872 PR.
I think this change is just for providing the latest react, a minimum 16.
(in the description of CRA: `react-scripts` has a missing peer dependency on `react`)

File: package.json [hunk 8]
```diff
@@ -39,64 +39,64 @@
...
+    "babel-loader": "^8.2.3",
+    "babel-plugin-dev-expression": "^0.2.2",
```
Comment (MikyungKim) [inline package.json:62]:
```suggestion
    "babel-plugin-dev-expression": "^0.2.3",
```

File: package.json [hunk 9]
```diff
@@ -39,64 +39,64 @@
...
+    "babel-plugin-dynamic-import-node": "^2.3.3",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
+    "babel-plugin-transform-rename-import": "^2.3.0",
+    "case-sensitive-paths-webpack-plugin": "^2.4.0",
+    "chalk": "^4.1.2",
+    "core-js": "^3.19.2",
```
Comment (MikyungKim) [inline package.json:68]:
```suggestion
    "core-js": "^3.21.1",
```

File: package.json [hunk 10]
```diff
@@ -39,64 +39,64 @@
...
+    "cross-spawn": "^7.0.3",
+    "css-loader": "^6.2.0",
```
Comment (MikyungKim) [inline package.json:70]:
```suggestion
    "css-loader": "^6.7.1",
```

File: package.json [hunk 11]
```diff
@@ -39,64 +39,64 @@
...
+    "babel-plugin-dev-expression": "^0.2.3",
+    "babel-plugin-dynamic-import-node": "^2.3.3",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
+    "babel-plugin-transform-rename-import": "^2.3.0",
+    "case-sensitive-paths-webpack-plugin": "^2.4.0",
+    "chalk": "^4.1.2",
+    "core-js": "^3.19.2",
+    "cross-spawn": "^7.0.3",
+    "css-loader": "^6.2.0",
+    "css-minimizer-webpack-plugin": "^3.2.0",
```
Comment (MikyungKim) [inline package.json:71]:
```suggestion
    "css-minimizer-webpack-plugin": "^3.4.1",
```

File: package.json [hunk 12]
```diff
@@ -39,64 +39,64 @@
...
+    "dotenv": "^10.0.0",
+    "dotenv-expand": "^5.1.0",
```
Comment (MikyungKim) [inline package.json:73]:
```suggestion
    "dotenv": "^16.0.0",
    "dotenv-expand": "^8.0.3",
```

File: package.json [hunk 13]
```diff
@@ -39,64 +39,64 @@
...
     "enzyme": "3.11.0",
-    "eslint": "7.30.0",
+    "eslint": "7.11.0",
```
Comment (MikyungKim) [inline package.json:75]:
```suggestion
    "eslint": "^7.32.0",
```

File: package.json [hunk 14]
```diff
@@ -39,64 +39,64 @@
...
     "eslint-config-enact": "3.1.3",
     "eslint-plugin-babel": "5.3.1",
     "eslint-plugin-enact": "0.2.3",
```
Comment (MikyungKim) [inline package.json:78]:
```suggestion
    "eslint-config-enact": "^3.1.3",
    "eslint-plugin-babel": "^5.3.1",
    "eslint-plugin-enact": "^0.2.3",
```

File: config/webpack.config.js (hunk 3 above)
Comment (hong6316) [reply config/webpack.config.js]: OK

File: package.json [hunk 15]
```diff
@@ -39,64 +39,64 @@
...
+    "html-webpack-plugin": "^5.5.0",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^27.4.3",
+    "jest-watch-typeahead": "^1.0.0",
+    "less": "^3.11.3",
+    "less-loader": "^6.2.0",
+    "less-plugin-npm-import": "^2.1.0",
+    "license-checker": "^25.0.1",
+    "mini-css-extract-plugin": "^2.1.0",
+    "minimist": "^1.2.5",
```
Comment (MikyungKim) [inline package.json:99]:
```suggestion
    "eslint-plugin-jsx-a11y": "^6.5.1",
    "eslint-plugin-react": "^7.29.4",
    "eslint-plugin-react-hooks": "^4.3.0",
    "eslint-plugin-testing-library": "^5.1.0",
    "eslint-webpack-plugin": "^3.1.1",
    "expose-loader": "3.1.0",
    "file-loader": "^6.2.0",
    "filesize": "^8.0.7",
    "fs-extra": "^10.0.1",
    "glob": "^7.2.0",
    "global-modules": "^2.0.0",
    "html-webpack-plugin": "^5.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^27.5.1",
    "jest-watch-typeahead": "^1.0.0",
    "less": "^4.1.2",
    "less-loader": "^10.2.0",
    "less-plugin-npm-import": "^2.1.0",
    "license-checker": "^25.0.1",
    "mini-css-extract-plugin": "^2.6.0",
    "minimist": "^1.2.5",
```

File: package.json [hunk 16]
```diff
@@ -105,27 +105,29 @@
...
+    "strip-ansi": "^6.0.1",
+    "style-loader": "^3.0.0",
+    "tailwindcss": "^3.0.2",
+    "tar": "^6.1.11",
+    "terser-webpack-plugin": "^5.2.5",
+    "v8-compile-cache": "^2.3.0",
+    "validate-npm-package-name": "^3.0.0",
+    "webpack": "^5.64.4",
+    "webpack-dev-server": "^4.6.0"
+
```
Comment (MikyungKim) [inline package.json:127]: please remove this line.

File: package.json [hunk 17]
```diff
@@ -39,64 +39,64 @@
...
+    "filesize": "^8.0.7",
+    "fs-extra": "^10.0.1",
+    "glob": "^7.2.0",
+    "global-modules": "^2.0.0",
+    "html-webpack-plugin": "^5.5.0",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^27.5.1",
+    "jest-watch-typeahead": "^1.0.0",
+    "less": "^4.1.2",
+    "less-loader": "^10.2.0",
```
Comment (MikyungKim) [inline package.json:95]:
```suggestion
    "less-loader": "^8.1.1",
```

**Review by MikyungKim — APPROVED** [review]
LGTM
